packages/sourcecred/src/plugins/discord/createGraph.js

```diff
@@ -320,48 +320,53 @@ export function _createGraphFromMessages(
       channelId,
       channelParentId,
     } = graphMessage;
-    if (author) {
-      wg.graph.addNode(memberNode(author));
-      wg.graph.addEdge(authorsMessageEdge(message, author));
-    }
-    wg.graph.addNode(messageNode(message, guildId, channelName));
 
+    let messageWeight = 0;
     for (const {reaction, reactingMember} of reactions) {
-      const node = reactionNode(reaction, message.timestampMs, guildId);
-      wg.weights.nodeWeights.set(
-        node.address,
-        reactionWeight(
-          weights,
-          message,
-          reaction,
-          reactingMember,
-          propsChannels,
-          reactions,
-          channelParentId
-        )
-      );
-      wg.graph.addNode(node);
-      wg.graph.addNode(memberNode(reactingMember));
-      wg.graph.addEdge(reactsToEdge(reaction, message));
-      wg.graph.addEdge(
-        addsReactionEdge(reaction, reactingMember, message.timestampMs)
-      );
+      const weight = reactionWeight({
+        weights,
+        message,
+        reaction,
+        reactingMember,
+        propsChannels,
+        reactions,
+        channelParentId,
+      });
+      messageWeight += weight;
+      if (weight && !config.simplifyGraph) {
+        const node = reactionNode(reaction, message.timestampMs, guildId);
+        wg.weights.nodeWeights.set(node.address, weight);
+        wg.graph.addNode(node);
+        wg.graph.addNode(memberNode(reactingMember));
+        wg.graph.addEdge(reactsToEdge(reaction, message));
+        wg.graph.addEdge(
+          addsReactionEdge(reaction, reactingMember, message.timestampMs)
+        );
+      }
     }
-
-    for (const {member, count} of mentions) {
-      wg.graph.addNode(memberNode(member));
-      let edge;
-      if (propsChannels.has(channelId)) {
-        edge = propsEdge(message, member);
-      } else {
-        edge = mentionsEdge(message, member);
+    if (messageWeight) {
```

Comment:
It's unclear to me why this is added here.  Is there a higher-level meaning we can assign to messageWeight, like `const hasNoWeight=!messageWeight`.  non-blocking

Reply:
...assuming that's the reasoning for the conditional

Reply:
... and if we can stand the incremental cost of another variable.  

else maybe `// don't bother adding weights to a message having no value` ??